### ISSUE TITLE
Invalid example in ReusingStepDefinitions.md

### ISSUE THREAD

**Issue by eternalmatt** [bencompton/jest-cucumber#65]
Please take look at [ReusingStepDefinitions.md](https://github.com/bencompton/jest-cucumber/blob/HEAD/docs/ReusingStepDefinitions.md) and notice that in the suggested example (second large code block) there is a variable `myAccount`.  However this variable does not exist at this point in time.  It seems this shared-steps was created as a result of hasty refactoring, and I see no way to link myAccount in the  .steps file to the shared-steps.

My suggestion would be create a class in shared-steps which can return steps when new'd.

```
import { factory } from 'shared-steps'
const { givenIHaveXDollarsInMyBankAccount} = new factory(myAccount);
```

**Comment by boboz2** [comment]
I also have a problem with the example ReusingStepDefinitions.md.
I follow the example but i have some error.

```
// shared-steps.js

export const givenIHaveXDollarsInMyBankAccount = given => {
    given(/I have \$(\d+) in my bank account/, balance => {
        myAccount.deposit(balance);
    });
};

export const thenMyBalanceShouldBe = then => {
    then(/my balance should be \$(\d+)/, balance => {
        expect(myAccount.balance).toBe(balance);
    });
};
```

```
// example.steps.js

import { loadFeature, defineFeature } from 'jest-cucumber';
import { thenMyBalanceShouldBe, givenIHaveXDollarsInMyBankAccount } from './shared-steps';

const feature = loadFeature('example.feature');

defineFeature(feature, test => {
  let myAccount;
		
  beforeEach(() => {
    myAccount = new BankAccount();
  });

  test('Making a deposit', ({ given, when, then }) => {
    givenIHaveXDollarsInMyBankAccount(given);

    when(/I deposit \$(\d+)/, deposit => {
      myAccount.deposit(deposit);
    });

    thenMyBalanceShouldBe(then);
  });
	
  test('Making a withdrawal', ({ given, when, then }) => {
    givenIHaveXDollarsInMyBankAccount(given);

    when(/I withdraw \$(\d+)/, withdrawal => {
      myAccount.withdraw(withdrawal);
    });		

    thenMyBalanceShouldBe(then);
  });
});
export class BankAccount {
  public balance: number = 0;

  public deposit(amount: number) {
      this.balance += amount;
  }

  public withdraw(amount: number) {
      this.balance -= amount;
  }
}
```
I have this error : 
![image](https://user-images.githubusercontent.com/32549972/65317944-13cc1100-db9d-11e9-9a60-ef258bef66b6.png)

**Comment by Kiyozz** [comment]
Same as @boboz2 !

**Comment by bencompton** [comment]
Yes, that example is quite flawed. What we need is an easy way to share state between files, as suggested in #61.

**Comment by aljacinto** [comment]
anything wrong with just passing the context? (other than it is kind of hack)

```
const givenExample = (given, context) => given("given example", async () => {
    const payload = 'hello';
    context = {payload};
});

const whenExample = (when, context) => when("when example", async () => {
    console.log(context.payload, 'world')
});

```

the only thing I can't figure out is if I have async call inside my method, I want the given to execute everything inside it before going to when.  Is that possible?

**Comment by yellowbrickc** [comment]
The state cannot really be "shared" because it belongs to the scenario but what about making the shared functions depending only on parameter and not on global states? It is a better coding style anyway ...

```js
// shared-steps.js
export const givenIHaveXDollarsInMyBankAccount = (given, account) => {
    given(/I have \$(\d+) in my bank account/, balance => {
        account.deposit(balance);
    });
};

// example.steps.js
givenIHaveXDollarsInMyBankAccount(given, myAccount);
```

What do you think?

**Comment by Kiyozz** [comment]
> The state cannot really be "shared" because it belongs to the scenario but what about making the shared functions depending only on parameter and not on global states? It is a better coding style anyway ...
> 
> ```js
> // shared-steps.js
> export const givenIHaveXDollarsInMyBankAccount = (given, account) => {
>     given(/I have \$(\d+) in my bank account/, balance => {
>         account.deposit(balance);
>     });
> };
> 
> // example.steps.js
> givenIHaveXDollarsInMyBankAccount(given, myAccount);
> ```
> 
> What do you think?

The documentation needs to be updated then. This is real better. I use a class to shared my steps and states instead of functions.

**Comment by yellowbrickc** [comment]
I can prepare a PR if you would like but it could take a few days. Should I or are you already working on it?

**Comment by Kiyozz** [comment]
I can’t work on it right now. I think you should do it 😉

**Comment by yellowbrickc** [comment]
@bencompton @Kiyozz the PR should be made for the master or for 2.0.12 ?

**Comment by yellowbrickc** [comment]
> @bencompton @Kiyozz the PR should be made for the master or for 2.0.12 ?

I merge it to default (master), it is an inoffensiv merge anyway.

**Comment by bencompton** [comment]
Fixed in #76
